docs/guide/api-plugin.md

```diff
@@ -625,16 +625,4 @@ export default defineConfig({
 
 ### TypeScript for Custom Events
 
-It is possible to type custom events by extending the `CustomEventMap` interface:
```

Comment:
> I also think this part of the docs would be a better fit in the HMR API section rather than Plugin API, since it's a customization of the custom event type used in HMR

I think it's more a Plugin API feature because a plugin that handles that event is necessary. On the other hand, other HMR API features doesn't require a plugin.

Reply:
Okay, would it still be fine to have a backlink from HMR API to Plugin API regarding "Typescript for Custom Events"?

Reply:
> See [Client-server Communication](/guide/api-plugin.html#client-server-communication) for more details.

We already have a link to the section. I guess we don't need it, but if the link was hard to find, I think we can add a bit more text that it also has a section about TypeScript (e.g. `See Client-server Communication for more details and TypeScript declaration`, not sure about the wording).